### PR TITLE
Fix for issue#2354 : Password length can now be upto 128 characters.

### LINKED ISSUE #2354: Password limit is only 10 characters.
**Actual Behaviour**

The maximum length of the password that can be entered by the user is 10 characters.

**Expected Behaviour**

The maximum length of password should be 128 characters according to [owasp](https://www.owasp.org/index.php/Authentication_Cheat_Sheet#Password_Length).

**Steps to reproduce it**

1) Open security settings to set password.
2) The maximum length of password is 10 characters.

**Would you like to work on the issue?**

YES.

### PR DESCRIPTION
**Pull request by yashk2000**
Fixed #2354 

Changes: Passwords now have upto 128 characters.

### CODE REVIEW

File: app/src/main/java/org/fossasia/phimpme/gallery/activities/SecurityActivity.java [hunk 1]
```diff
@@ -270,9 +270,9 @@ private void setPasswordDialog() {
             }
 
             @Override public void afterTextChanged(Editable editable) {
-                if(editable.length() == 11) {
-                    editTextConfirmPassword.setText(editable.toString().substring(0, 10));
-                    editTextConfirmPassword.setSelection(10);
+                if(editable.length() == 128) {
```
[the remark this answers is not in the document]
Comment (yashk2000) [reply app/src/main/java/org/fossasia/phimpme/gallery/activities/SecurityActivity.java]: Right now if the length is greater than 128 then a toast is shown and any extra characters typed by the user are not shown. Should I still change it to greater than?

[the remark this answers is not in the document]
Comment (yashk2000) [reply app/src/main/java/org/fossasia/phimpme/gallery/activities/SecurityActivity.java]: @abishekvashok this is what happens right now if 128 characters are reached:

different

Please tell whether I should make changes or not?

Comment (frextrite) [inline app/src/main/java/org/fossasia/phimpme/gallery/activities/SecurityActivity.java]: This checks if the total characters entered are exactly 128, it then shows a pop-up displaying maximum password length. But what if someone pastes a password which is greater than 128 characters, then this statement would never be called, and the toast would never be shown.

Comment (yashk2000) [reply app/src/main/java/org/fossasia/phimpme/gallery/activities/SecurityActivity.java]: @frextrite If someone pastes a password which is more than 128 characters long, the result would be the same. A toast is displayed and only the first 128 characters of the pasted text are shown.

**Comment by yashk2000** [comment]
@frextrite I would have used a variable if the maximum length of the password was varying. I have fixed the maximum length to be 128 characters which cannot be varied so there isn't a need to use a variable.

File: app/src/main/java/org/fossasia/phimpme/gallery/activities/SecurityActivity.java (hunk 1 above)
Comment (frextrite) [reply app/src/main/java/org/fossasia/phimpme/gallery/activities/SecurityActivity.java]: That's great then!

**Comment by frextrite** [comment]
Although you are right that max length value is fixed at ```128``` and won't change at runtime, but it's just good practice to use a variable, declare its value once and use it everywhere. It makes the code more readable, and easy to mantain. In this case, a ```const``` variable would do the job.

**Comment by yashk2000** [comment]
@frextrite that's right. I'll make the change.

**Comment by abishekvashok** [comment]
Thanks

**Pull request merged**